app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt

```diff
@@ -40,7 +40,12 @@ class TabsSettingsFragment : PreferenceFragmentCompat() {
     }
 
     private fun setupPreferences() {
-        listRadioButton = requirePreference(R.string.pref_key_tab_view_list)
+        // This should be the only use case for pref_key_tab_view_list_do_not_use.
+        // In the Fenix impl of RadioGroups, we need to always have an individual pref value
+        // for it to work. This is weird for a radio group that should hold a value from that group.
+        // For the tabs tray, we only need a boolean value, so let's rely on only the
+        // pref_key_tab_view_grid and look into using the native RadioGroup in the future.
+        listRadioButton = requirePreference(R.string.pref_key_tab_view_list_do_not_use)
```

Comment:
This one off seems a bit weird no matter how many times I look at it. I am wondering if it would just be better to keep with the existing pattern. I assume `pref_key_tab_view_list` will still flip to the appropriate boolean value it needs and can still be used correctly in `TabsTouchHelper`. This might be better explored in a followup issue to use native RadioGroup (let's file an issue for this if it doesn't already exist).

I can see the argument that we just want to get users into using only one preference for this since this is the new default as a motivation for doing this.

Aside we also just added another radio group in https://github.com/mozilla-mobile/fenix/pull/19791, which is why I feel conflicted about this weirdness - one side it makes total sense to just use one preference, but it's also so icky to see a pref that you should not be using at all.

Reply:
I tried looking into it when I put this PR, but it's turtles all the way down: a radio button is meant to be a selected value based on group of options. I didn't want to change the world for this feature PR, but I wanted to remove the concept of two booleans to represent one. I removed all other references to the list pref (which was thankfully just the one now that we deleted the old code), and renamed the string ID so we don't mistakenly use it again.

I agree we need to fix the underlying problem, rather than the individual features. I'll file a bug for this to discuss it there. No need to block #19791 about this either until we have a good solution.

Reply:
Filed as https://github.com/mozilla-mobile/fenix/issues/19829.